addon/-private/system/diff-array.js

```diff
@@ -0,0 +1,60 @@
+/**
+  @namespace
+  @method diff-array
+  @for DS
+  @param {Array} oldArray the old array
+  @param {Array} newArray the new array
+  @return {hash} {
+      firstChangeIndex: <integer>,  // null if no change
+      addedCount: <integer>,        // 0 if no change
+      removedCount: <integer>       // 0 if no change
+    }
+*/
+const diffArray = function (oldArray, newArray) {
```

Comment:
do you mind if we do `export default function diffArray`

* This lets readers easily see (without scanning the full file) that this method is the default export